machineconfiguration/v1alpha1/types_osimagestream.go

```diff
@@ -0,0 +1,124 @@
+	// Must reference the name of one of the streams in availableStreams.
+	// Must be set when availableStreams is not empty.
+	// When not set and availableStreams is empty, controllers should use the default one stated in the release image.
+	//
+	// +optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=70
+	// +kubebuilder:validation:XValidation:rule=`self.matches('^[\\w\\.\\-]+$')`,message="The name must consist only of alphanumeric characters, hyphens ('-') and dots ('.')."
+	DefaultStream string `json:"defaultStream,omitempty"`
+}
+
+// OSImageStreamSpec defines the desired state of a OSImageStream.
+type OSImageStreamSpec struct {
+}
+
+type OSImageStreamURLSet struct {
+	// name is the identifier of the stream.
+	//
+	// Must not be empty and must not exceed 70 characters in length.
+	// Must only contain alphanumeric characters, hyphens ('-'), or dots ('.').
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=70
+	// +kubebuilder:validation:XValidation:rule=`self.matches('^[\\w\\.\\-]+$')`,message="The name must consist only of alphanumeric characters, hyphens ('-') and dots ('.')."
+	Name string `json:"name,omitempty"`
+
+	// osImageUrl is an OS Image referenced by digest.
+	//
+	// The format of the URL ref is:
+	// host[:port][/namespace]/name@sha256:<digest>
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=447
+	// +kubebuilder:validation:XValidation:rule=`self.split('@').size() == 2 && self.split('@')[1].matches('^sha256:[a-f0-9]{64}$')`,message="the OCI Image reference must end with a valid '@sha256:<digest>' suffix, where '<digest>' is 64 characters long"
+	// +kubebuilder:validation:XValidation:rule=`self.split('@')[0].matches('^([a-zA-Z0-9-]+\\.)+[a-zA-Z0-9-]+(:[0-9]{2,5})?/([a-zA-Z0-9-_]{0,61}/)?[a-zA-Z0-9-_.]*?$')`,message="the OCI Image name should follow the host[:port][/namespace]/name format, resembling a valid URL without the scheme"
+	OSImageUrl string `json:"osImageUrl,omitempty"`
+
+	// osExtensionsImageUrl is an OS Extensions Image referenced by digest.
+	//
+	// The format of the URL ref is:
+	// host[:port][/namespace]/name@sha256:<digest>
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=447
+	// +kubebuilder:validation:XValidation:rule=`self.split('@').size() == 2 && self.split('@')[1].matches('^sha256:[a-f0-9]{64}$')`,message="the OCI Image reference must end with a valid '@sha256:<digest>' suffix, where '<digest>' is 64 characters long"
+	// +kubebuilder:validation:XValidation:rule=`self.split('@')[0].matches('^([a-zA-Z0-9-]+\\.)+[a-zA-Z0-9-]+(:[0-9]{2,5})?/([a-zA-Z0-9-_]{0,61}/)?[a-zA-Z0-9-_.]*?$')`,message="the OCI Image name should follow the host[:port][/namespace]/name format, resembling a valid URL without the scheme"
+	OSExtensionsImageUrl string `json:"osExtensionsImageUrl,omitempty"`
```

Comment:
thought (non-blocking): Would it make sense for us to put the `coreos-bootimages` ConfigMap content in here as well? We could include it the same way that we include Ignition content in MachineConfigs by making the field type a `runtime.RawExtension`.